core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -50,6 +57,37 @@ object ZkUtils extends Logging {
   val EntityConfigPath = "/config"
   val EntityConfigChangesPath = "/config/changes"
 
+  /** true if java.security.auth.login.config is set to some jaas file which has "Client" entry. **/
+
+  val isSecure: Boolean = {
+    val loginConfigurationFile: String = System.getProperty("java.security.auth.login.config")
```

Comment:
Yes, both options have drawbacks. I think there's a benefit in being able to use `server.properties` instead of the system property, but it is true that it introduces some complexity. I'll leave it up to @gwenshap and/or @junrao to give their opinions.